overlord/snapstate/handlers.go

```diff
@@ -1548,9 +1663,41 @@ func (m *SnapManager) stopSnapServices(t *state.Task, _ *tomb.Tomb) error {
 
 	pb := NewTaskProgressAdapterUnlocked(t)
 	st.Unlock()
+	defer st.Lock()
+
+	// stop the services
 	err = m.backend.StopServices(svcs, stopReason, pb, perfTimings)
```

Comment:
same comment about st.Unlock()/std.Lock() here

Reply:
As we discussed on IRC, what I have is a bit more readable and safer against future refactors, so I will leave this as is for now.